srsenb/src/stack/upper/s1ap.cc

```diff
@@ -982,6 +1048,92 @@ bool s1ap::send_initial_ctxt_setup_failure(uint16_t rnti)
   return true;
 }
 
+bool s1ap::send_uectxmodifyresp(uint16_t rnti)
+{
+  if(!mme_connected) {
+    return false;
+  }
+  srslte::unique_byte_buffer_t buf = srslte::allocate_unique_buffer(*pool);
+  if (!buf) {
+    s1ap_log->error("Fatal Error: Couldn't allocate buffer in s1ap::send_uectxmodifyresp().\n");
+    return false;
+  }
+
+  LIBLTE_S1AP_S1AP_PDU_STRUCT tx_pdu;
+  tx_pdu.ext          = false;
+  tx_pdu.choice_type  = LIBLTE_S1AP_S1AP_PDU_CHOICE_SUCCESSFULOUTCOME;
+
+  LIBLTE_S1AP_SUCCESSFULOUTCOME_STRUCT *succ = &tx_pdu.choice.successfulOutcome;
+  succ->procedureCode = LIBLTE_S1AP_PROC_ID_UECONTEXTMODIFICATION;
+  succ->choice_type   = LIBLTE_S1AP_SUCCESSFULOUTCOME_CHOICE_UECONTEXTMODIFICATIONRESPONSE;
+
+  LIBLTE_S1AP_MESSAGE_UECONTEXTMODIFICATIONRESPONSE_STRUCT *resp = &succ->choice.UEContextModificationResponse;
+  resp->ext                             = false;
+  resp->CriticalityDiagnostics_present  = false;
+
+  resp->MME_UE_S1AP_ID.MME_UE_S1AP_ID = ue_ctxt_map[rnti].MME_UE_S1AP_ID;
+  resp->eNB_UE_S1AP_ID.ENB_UE_S1AP_ID = ue_ctxt_map[rnti].eNB_UE_S1AP_ID;
+
+  liblte_s1ap_pack_s1ap_pdu(&tx_pdu, (LIBLTE_BYTE_MSG_STRUCT*)buf.get());
+  s1ap_log->info_hex(buf->msg, buf->N_bytes, "Sending ContextModificationFailure for RNTI:0x%x", rnti);
+
+  ssize_t n_sent = sctp_sendmsg(socket_fd, buf->msg, buf->N_bytes,
+                                (struct sockaddr*)&mme_addr, sizeof(struct sockaddr_in),
+                                htonl(PPID), 0, ue_ctxt_map[rnti].stream_id, 0, 0);
+
+  if(n_sent == -1) {
+    s1ap_log->error("Failed to send ContextModificationFailure for RNTI:0x%x\n", rnti);
+    return false;
+  }
+
+  return true;
+}
+
+
+bool s1ap::send_uectxmodifyfailure(uint16_t rnti, LIBLTE_S1AP_CAUSE_STRUCT *cause)
+{
+  if(!mme_connected) {
```

Comment:
same, and in a few other places below

Reply:
I added the script  you referenced as git pre-commit hook, and then did a rebase -i, it didn't  flag anything.  I also manually edited the patch and rebased/edited it a number of times - at no point did the hook flag anything as being incorrect.

In absence of clear guidelines on how actually to check in an automatic/efficient way, I manually looked at the diff and opened each file in vim, navitgated to the sections of code added, marke them in visual mode and then C-K to execute the clang format.

If you insist on suprer-correct formatting in all cases, I strongly recommend that you provide proper tooling and a very clear contribution guide for people to determine if it's correct or not.

Reply:
No idea why the pre-commit hook didn't catch the wrongly formatted lines in your case. I think a `rebase -i` should also detect them. In any case, the formatting looks ok now and we can merge the commit. Thanks